docker/models/containers.py

```diff
@@ -779,7 +779,7 @@ def run(self, image, command=None, stdout=True, stderr=False,
         out = None
         if logging_driver == 'json-file' or logging_driver == 'journald':
             out = container.logs(
-                stdout=stdout, stderr=stderr, stream=True, follow=True
+                stdout=stdout, stderr=stderr, stream=True, follow=False
```

Comment:
This change concerns me because it means we're losing any logs that happen after the method is called. I don't think that is correct.

We might want to use `attach` instead?

Reply:
I guess it's easier to call container.logs after container is stopped. What about this:

```diff
--- a/docker/models/containers.py
+++ b/docker/models/containers.py
@@ -777,16 +777,15 @@ class ContainerCollection(Collection):
         logging_driver = container.attrs['HostConfig']['LogConfig']['Type']
 
         out = None
-        if logging_driver == 'json-file' or logging_driver == 'journald':
-            out = container.logs(
-                stdout=stdout, stderr=stderr, stream=True, follow=False
-            )
-
         exit_status = container.wait()['StatusCode']
         if exit_status != 0:
-            out = None
             if not kwargs.get('auto_remove'):
                 out = container.logs(stdout=False, stderr=True)
+        else:
+            if logging_driver == 'json-file' or logging_driver == 'journald':
+                out = container.logs(
+                    stdout=stdout, stderr=stderr, stream=True, follow=False
+                )
 
         if remove:
             container.remove()
```

Reply:
One thing I am not sure is how it works (and worked before) in case `remove=True`.

Reply:
That change looks fine to me.

Reply:
Yeah, `container.logs()` after container stop won't work in case of `auto_remove=True`.

We can use `container.logs` with `follow=True` in a separate thread and kill it once the container is stopped. The problem is I am not quite sure how to implement it in Python :(

Reply:
Or maybe we just need to do `out.Close()` in proper places...